File: API/Data/Entities/Order.cs
using System;
using System.Collections.Generic;

namespace API.Data.Entities
{
    public partial class Order
    {
        public Order()
        {
            OrderItems = new HashSet<OrderItem>();
            Payments = new HashSet<Payment>();
        }

        public string? Note { get; set; }
        public decimal Amount { get; set; }
        public short Status { get; set; }
        public DateTime? CreateAt { get; set; }
        public DateTime? UpdateAt { get; set; }
        public int? PaymentId { get; set; }
        public string Address { get; set; } = null!;
        public string Fullname { get; set; } = null!;
        public string PhoneNumber { get; set; } = null!;
        public int ShippingFee { get; set; }
        public string Email { get; set; } = null!;
        public short DeliveryMethod { get; set; }
        public Guid? UserId { get; set; }
        public decimal Subtotal { get; set; }
        public decimal DiscountAmount { get; set; }
        public int? PromotionId { get; set; }
        public Guid Id { get; set; }

        public virtual User? User { get; set; }
        public virtual ICollection<OrderItem> OrderItems { get; set; }
        public virtual ICollection<Payment> Payments { get; set; }
    }
}

File: Client/src/app/shared/models/order.ts
export interface OrderRequest {
  fullname: string;
  email: string;
  phoneNumber: string;
  address: string;
  deliveryMethod: number;
  paymentMethod: string;
  shippingFee: number;
  amount: number;
  subtotal: number;
  discountAmount: number;
  promotionCode: string;
  note: string;
}

export interface Order {
  id: string;
  amount: number;
  status: number;
  createAt: string;
  updateAt: any;
  address: string;
  fullname: string;
  phoneNumber: string;
  paymentMethod: string;
}

export interface OrderDetail {
  id: string
  amount: number
  subtotal: number
  discountAmount: number
  shippingFee: number
  fullname: string
  email: string
  address: string
  phoneNumber: string
  note: string
  status: number
  deliveryMethod: number
  createAt: string
  updateAt: any
  paymentInfo: PaymentInfo
  orderItems: OrderItem[]
}

export interface OrderItem {
  id: number
  quantity: number
  productName: string
  productColor: string
  productSize: string
  price: number
  discount: number
  productImageUrl: string
}

export interface PaymentInfo {
  method: string
  provider: string
  amount: number
  createAt: string
  updateAt: string
  transactionId: string
  userId: any
  status: boolean
  currencyCode: string
}

export enum OrderStatus {
  Pending = 0,
  Confirmed = 1,
  Shipping = 2,
  Completed = 3,
  Cancelled = 4
}
